tests/ecs/ambiguity_detection.rs

```diff
@@ -0,0 +1,94 @@
+//! An example to confirm that `bevy` doesn't have system order ambiguity with [`DefaultPlugins`]
+//! This is run in CI to ensure that this doesn't regress again.
+
+use bevy::{
+    ecs::schedule::{LogLevel, ScheduleBuildSettings, ScheduleLabel},
+    prelude::*,
+};
+use bevy_render::{pipelined_rendering::RenderExtractApp, Render, RenderApp};
+
+/// FIXME: bevy should not have any ambiguities, but it takes time to clean these up,
+/// so we're juste ignoring those for now.
+///
+/// See [#7386](https://github.com/bevyengine/bevy/issues/7386) for relevant issue.
+pub fn get_ignored_ambiguous_systems() -> Vec<Box<dyn ScheduleLabel>> {
+    vec![
+        Box::new(First),
+        Box::new(PreUpdate),
+        Box::new(Update),
+        Box::new(PostUpdate),
+        Box::new(Last),
+        Box::new(ExtractSchedule),
+        Box::new(Render),
+    ]
+}
+
+/// A test to confirm that `bevy` doesn't have system order ambiguity with [`DefaultPlugins`]
+/// This is run in CI to ensure that this doesn't regress again.
+pub fn main() {
+    let mut app = App::new();
+    app.add_plugins(DefaultPlugins);
+
+    fn configure_ambiguity_detection(sub_app: &mut SubApp) {
+        let ignored_ambiguous_systems = get_ignored_ambiguous_systems();
+        let mut schedules = sub_app.world_mut().resource_mut::<Schedules>();
+        for (_, schedule) in schedules.iter_mut() {
+            if ignored_ambiguous_systems
+                .iter()
+                .any(|label| **label == *schedule.label())
+            {
+                continue;
+            }
+            schedule.set_build_settings(ScheduleBuildSettings {
+                ambiguity_detection: LogLevel::Warn,
+                use_shortnames: false,
+                ..default()
+            });
+        }
+    }
+    let sub_app = app.main_mut();
+    configure_ambiguity_detection(sub_app);
+    let sub_app = app.sub_app_mut(RenderApp);
+    configure_ambiguity_detection(sub_app);
+    let sub_app = app.sub_app_mut(RenderExtractApp);
+    configure_ambiguity_detection(sub_app);
+
+    app.finish();
+    app.cleanup();
+    app.update();
+
+    /// Returns the number of conflicting systems.
+    fn assert_no_conflicting_systems(sub_app: &SubApp) -> usize {
```

Comment:
Some notes on this function
- Could you please move it out of the `main` function?
- The name is a total misnomer. The `no` should probably be `no_new`, but the what the function actually does is twofold
    - Assert that there are no unknown ambiguities
    - Count the amount of known ambiguities
    - A clean solution would probably be to return a `struct` containing the amount of known and unknown ambiguities. Then, the calling code can run the `assert`s on these two values.

Reply:
I did my best to provide a clean solution, there's `.0` around but the result is better than before, with more exhaustive report rather than failing at the first unexpected ambiguity detected.

new report example (for a fail test):
```sh
1 pairs of systems with conflicting data access have indeterminate execution order. Consider adding `before`, `after`, or `ambiguous_with` relationships between these:
 -- bevy_app::terminal_ctrl_c_handler::TerminalCtrlCHandlerPlugin::exit_on_flag and bevy_dev_tools::ci_testing::systems::send_events
    conflict on: bevy_ecs::world::World


thread 'main' panicked at tests/ecs/ambiguity_detection.rs:58:5:
assertion `left == right` failed: Main app has unexpected ambiguities among these schedules: [Update].
More Details:
AmibiguitiesCount(
    {
        FixedPostUpdate: 0,
        FixedLast: 0,
        StateTransition: 0,
        RunFixedMainLoop: 0,
        FixedMain: 0,
        PostUpdate: 64,
        FixedFirst: 0,
        Last: 3,
        PostStartup: 0,
        First: 1,
        Update: 1,
        SpawnScene: 0,
        Main: 0,
        PreStartup: 0,
        PreUpdate: 13,
    },
)
  left: 1
 right: 0
```